src/sage/combinat/posets/lattices.py

```diff
@@ -734,7 +734,10 @@ def LatticePoset(data=None, *args, **options):
                 error.x = P._vertex_to_element(error.x)
                 error.y = P._vertex_to_element(error.y)
                 raise
-    return FiniteLatticePoset(P, category=FiniteLatticePosets(), facade=P._is_facade)
+    cat = options.get('category', None)
+    if cat is None:
+        cat = FiniteLatticePosets()
```

Comment:
```suggestion
    cat = Posets().or_subcategory(options.get('category', None))
    cat = cat & FiniteLatticePosets()
```
I think this is better since it protects against a user passing something nonsensical.